front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -1,20 +1,34 @@
 import { useEffect, useState } from 'react';
 
+import { useTranslation } from 'react-i18next';
+
 import { STDCM_TRAIN_ID } from 'applications/stdcm/consts';
 import {
   osrdEditoastApi,
   type PathProperties,
   type PostV2InfraByInfraIdPathPropertiesApiArg,
 } from 'common/api/osrdEditoastApi';
+import { addElementAtIndex } from 'utils/array';
 
 const useGetProjectedTrainOperationalPoints = (
   trainIdUsedForProjection?: number,
   infraId?: number
 ) => {
+  const { t } = useTranslation('simulation');
   const [operationalPoints, setOperationalPoints] = useState<
     NonNullable<PathProperties['operational_points']>
   >([]);
 
+  const { data: trainScheduleUsedForProjection } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
+      {
+        id: trainIdUsedForProjection as number,
```

Comment:
Nit: in general I prefer to avoid `as` casts if possible, because it makes the compiler ignore any type mismatch. If `trainIdUsedForProjection` happens to be a `string` for instance because there is a mistake in the code, this cast would hide the issue because the type-checker would remain silent.

Here `trainIdUsedForProjection` is a `number | undefined` so we should be able to use `trainIdUsedForProjection!` instead of `trainIdUsedForProjection as number` to retain the type check.

Reply:
I used that syntax because it's the one we use in most of the app but I'm not specially attached to it, so I can change it :)

Reply:
Yeah. I'd really like to scan all of the frontend and drop any unnecessary cast that can be simplified.

For me as a reviewer, `as` is a red flag shouting "we think we're smarter than the compiler" and I always wonder if I'm really smarter :P (sometimes it's really necessary, sometimes not)

Reply:
Totally agreed :)